src/core/public/legacy/legacy_service.ts

```diff
@@ -55,6 +57,8 @@ export class LegacyPlatformService {
   public readonly legacyId = Symbol();
   private bootstrapModule?: BootstrapModule;
   private targetDomElement?: HTMLElement;
+  private readonly startDependencies$ = new Subject<[LegacyCoreStart, object]>();
+  private readonly startDependencies = this.startDependencies$.pipe(first()).toPromise();
```

Comment:
NIT: We could probably use a plain Promise instead of relying on a Subject here. May be better?

```typescript
constructor(private readonly params: LegacyPlatformParams) {
    this.startDependencies = new Promise({resolve} => { this.resolveStartDependencies = resolve; })
}

L82
(unchanged) getStartServices: () => this.startDependencies,

L128
this.resolveStartDependencies([legacyCore, plugins]);

```

Reply:
I went back and forth on this. While the promise approach is a bit less fancy, it requires that the resolve function is an optional and writable property, which I don't love:

```ts
class LegacyService {
  private resolveStartDependencies?: (startDeps: [LegacyCoreStart, object]) => void;
  public readonly startDependencies: new Promise({resolve} => { this.resolveStartDependencies = resolve; })

  // in #start() 

  this.resolveStartDependencies!([legacyCore, plugins]);
}
```

Small tradeoffs with either option. If you still prefer this way, I can change it.

Reply:
Good point `new Promise({resolve} => { this.resolveStartDependencies = resolve; })` makes the resolver optional. Guess a Subject is fine then